event-based-simulators/extras/GenerateMeasurements.py

```diff
@@ -0,0 +1,248 @@
+        log.info(
+            f"Found device '{device.name}', id: #{device.id}, owned by {device.owner}, number of children: {len(device.child_devices)}, type: {device.type}")
+        for childDevice in device.child_devices:
+            deviceExternalId, deviceExternalIdType = CheckDeviceExternalIdById(childDevice.id, c8y.base_url)
+            if not deviceExternalId:
+                return
+            if IsExternalIdTypeEventBasedSimulatorProfile(deviceExternalIdType):
+                filePath = CreateFilePath(Id=deviceExternalId)
+            else:
+                return
+            listOfChildDevicesExternalIds.append(deviceExternalId)
+            ExportMeasurements(filePath=filePath)
+    if deviceInTenantCount == 0:
+        log.info(f"No device in tenant {c8y.tenant_id} found")
+
+
+def GetExternalIdReponse(deviceId, baseUrl):
+    externalIdResponse = requests.get(f'{baseUrl}/identity/globalIds/{deviceId}/externalIds',
+                                      headers=C8Y_HEADERS)
+    if not externalIdResponse.ok:
+        log.error(externalIdResponse.json())
+        sys.exit()
+    else:
+        return externalIdResponse
+
+
+def CheckDeviceExternalIdById(deviceId, baseUrl):
+    externalIdResponse = GetExternalIdReponse(deviceId, baseUrl)
+
+    try:
+        deviceExternalId = externalIdResponse.json()['externalIds'][0]['externalId']
+        deviceExternalIdType = externalIdResponse.json()['externalIds'][0]['type']
+        log.info(
+            f"Found external id: {deviceExternalId} with type: {deviceExternalIdType} for the device with id {deviceId}")
+    except:
+        log.info(f"Could not find external id for the device with id {deviceId}")
+        return None, None
+
+    return deviceExternalId, deviceExternalIdType
+
+
+def IsExternalIdTypeEventBasedSimulatorProfile(deviceExternalIdType):
+    if deviceExternalIdType == C8Y_PROFILE_GROUP:
+        return True
+    else:
+        log.info(f"The type {deviceExternalIdType} of external ID must match with type {C8Y_PROFILE_GROUP}")
+        return False
+
+
+def CreateFilePath(Id):
+    # Check if folder containing data files exists and make one if not
+    if not os.path.exists('generated_data'):
+        os.makedirs('generated_data')
+    relativeFilePath = f'generated_data\{Id}.json'
+    filePath = os.path.join(os.path.dirname(__file__), relativeFilePath)
+    log.debug(f"Created successfully file path: {filePath}")
+    return filePath
+
+
+def ExportMeasurements(filePath):
+    jsonMeasurementsList = JsonMeasurementList(device_id=123456, distribution='normal')
+    AppendDataToJsonFile(jsonMeasurementsList, filePath, 'measurements')
+
+
+def JsonMeasurementList(device_id, distribution):
+    time_pointer = sim_time
+    time_count = 0
+    jsonMeasurementList = []
+    storageMeasurementDict = CreateStorageMeasurementDict()
+    storageMeasurementDict_1800s = CreateStorageMeasurementDict()
+    while time_pointer <= sim_end_time:
+        time_count += 1
```

Comment:
I dont get the time_count logik in this script. What is it about?

Reply:
time_count helps me to identify when the measurements for 1800s and 3600s should be created. Normally, only 600s measurements are made.